dask/delayed.py

```diff
@@ -230,7 +231,14 @@ def tokenize(*args, **kwargs):
 
 
 @curry
-def delayed(obj, name=None, pure=None, nout=None, traverse=True):
+def delayed(
+    obj: Any,
+    name: Hashable = None,
+    pure: Optional[bool] = None,
+    nout: Optional[int] = None,
```

Comment:
this is redundant.
```suggestion
    pure: bool = None,
    nout: int = None,
```

Reply:
@crusaderky explicitly using `Optional` is actually now considered best practice: https://stackoverflow.com/a/49724148 / https://www.python.org/dev/peps/pep-0484/#union-types